### ISSUE TITLE
Postgres Export Error

### ISSUE THREAD

**Issue by jeremytiki** [barseghyanartur/django-fobi#191]
I've run into an issue trying to export data from FOBI using a Postgres Database:

```
Request Method: | POST
-- | --
http://########/admin/fobi_contrib_plugins_form_handlers_db_store/savedformdataentry/?form_entry__id__exact=12
1.11.5
ProgrammingError
SELECT DISTINCT ON expressions must match initial ORDER BY expressions LINE 1: SELECT DISTINCT ON ("db_store_savedformdataentry"."form_data...                             ^
/srv/django-site/venv/lib/python3.4/site-packages/django/db/backends/utils.py in execute, line 65
/usr/sbin/uwsgi
3.4.5
```

```
Environment:


Request Method: POST
Request URL: http://#################/admin/fobi_contrib_plugins_form_handlers_db_store/savedformdataentry/?form_entry__id__exact=12

Django Version: 1.11.5
Python Version: 3.4.5
Installed Applications:
('django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.admin',
 'django.contrib.sites',
 'django.contrib.sitemaps',
 'django.contrib.staticfiles',
 'django.contrib.messages',
 'fsu',
 'corsheaders',
 'django_cleanup',
 'django_python3_ldap',
 'fobi',
 'fobi.contrib.themes.foundation5',
 'fobi.contrib.plugins.form_elements.fields.boolean',
 'fobi.contrib.plugins.form_elements.fields.checkbox_select_multiple',
 'fobi.contrib.plugins.form_elements.fields.date',
 'fobi.contrib.plugins.form_elements.fields.date_drop_down',
 'fobi.contrib.plugins.form_elements.fields.datetime',
 'fobi.contrib.plugins.form_elements.fields.decimal',
 'fobi.contrib.plugins.form_elements.fields.email',
 'fobi.contrib.plugins.form_elements.fields.file',
 'fobi.contrib.plugins.form_elements.fields.float',
 'fobi.contrib.plugins.form_elements.fields.hidden',
 'fobi.contrib.plugins.form_elements.fields.input',
 'fobi.contrib.plugins.form_elements.fields.integer',
 'fobi.contrib.plugins.form_elements.fields.ip_address',
 'fobi.contrib.plugins.form_elements.fields.null_boolean',
 'fobi.contrib.plugins.form_elements.fields.password',
 'fobi.contrib.plugins.form_elements.fields.radio',
 'fobi.contrib.plugins.form_elements.fields.regex',
 'fobi.contrib.plugins.form_elements.fields.select',
 'fobi.contrib.plugins.form_elements.fields.select_model_object',
 'fobi.contrib.plugins.form_elements.fields.select_multiple',
 'fobi.contrib.plugins.form_elements.fields.select_multiple_model_objects',
 'fobi.contrib.plugins.form_elements.fields.slug',
 'fobi.contrib.plugins.form_elements.fields.text',
 'fobi.contrib.plugins.form_elements.fields.textarea',
 'fobi.contrib.plugins.form_elements.fields.time',
 'fobi.contrib.plugins.form_elements.fields.url',
 'fobi.contrib.plugins.form_elements.test.dummy',
 'easy_thumbnails',
 'fobi.contrib.plugins.form_elements.content.content_image',
 'fobi.contrib.plugins.form_elements.content.content_image_url',
 'fobi.contrib.plugins.form_elements.content.content_text',
 'fobi.contrib.plugins.form_elements.content.content_video',
 'fobi.contrib.plugins.form_handlers.db_store',
 'fobi.contrib.plugins.form_handlers.http_repost',
 'fobi.contrib.plugins.form_handlers.mail',
 'guardian',
 'graphene_django',
 'fobi_forms_extension',
 'admissions_appeal',
 'ldap',
 'authenticate',
 'departments')
Installed Middleware:
('corsheaders.middleware.CorsMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'corsheaders.middleware.CorsPostCsrfMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.locale.LocaleMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware')



Traceback:

File "/srv/django-site/venv/lib/python3.4/site-packages/django/db/backends/utils.py" in execute
  65.                 return self.cursor.execute(sql, params)

The above exception (SELECT DISTINCT ON expressions must match initial ORDER BY expressions
LINE 1: SELECT DISTINCT ON ("db_store_savedformdataentry"."form_data...
                            ^
) was the direct cause of the following exception:

File "/srv/django-site/venv/lib/python3.4/site-packages/django/core/handlers/exception.py" in inner
  41.             response = get_response(request)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/contrib/admin/options.py" in wrapper
  551.                 return self.admin_site.admin_view(view)(*args, **kwargs)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/utils/decorators.py" in _wrapped_view
  149.                     response = view_func(request, *args, **kwargs)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/views/decorators/cache.py" in _wrapped_view_func
  57.         response = view_func(request, *args, **kwargs)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/contrib/admin/sites.py" in inner
  224.             return view(request, *args, **kwargs)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/utils/decorators.py" in _wrapper
  67.             return bound_func(*args, **kwargs)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/utils/decorators.py" in _wrapped_view
  149.                     response = view_func(request, *args, **kwargs)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/utils/decorators.py" in bound_func
  63.                 return func.__get__(self, type(self))(*args2, **kwargs2)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/contrib/admin/options.py" in changelist_view
  1568.                 response = self.response_action(request, queryset=cl.get_queryset(request))

File "/srv/django-site/venv/lib/python3.4/site-packages/django/contrib/admin/options.py" in response_action
  1286.             response = func(self, request, queryset)

File "/srv/django-site/venv/lib/python3.4/site-packages/fobi/contrib/plugins/form_handlers/db_store/admin.py" in export_data
  65.         return data_exporter.graceful_export()

File "/srv/django-site/venv/lib/python3.4/site-packages/fobi/contrib/plugins/form_handlers/db_store/helpers.py" in graceful_export
  180.             return self._export_to_xls()

File "/srv/django-site/venv/lib/python3.4/site-packages/fobi/contrib/plugins/form_handlers/db_store/helpers.py" in _export_to_xls
  100.         data_headers = self._get_data_headers()

File "/srv/django-site/venv/lib/python3.4/site-packages/fobi/contrib/plugins/form_handlers/db_store/helpers.py" in _get_data_headers
  62.             qs = [obj.form_data_headers for obj in qs]

File "/srv/django-site/venv/lib/python3.4/site-packages/django/db/models/query.py" in __iter__
  250.         self._fetch_all()

File "/srv/django-site/venv/lib/python3.4/site-packages/django/db/models/query.py" in _fetch_all
  1118.             self._result_cache = list(self._iterable_class(self))

File "/srv/django-site/venv/lib/python3.4/site-packages/django/db/models/query.py" in __iter__
  53.         results = compiler.execute_sql(chunked_fetch=self.chunked_fetch)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/db/models/sql/compiler.py" in execute_sql
  894.             raise original_exception

File "/srv/django-site/venv/lib/python3.4/site-packages/django/db/models/sql/compiler.py" in execute_sql
  884.             cursor.execute(sql, params)

File "/srv/django-site/venv/lib/python3.4/site-packages/graphene_django/debug/sql/tracking.py" in execute
  154.         return self._record(self.cursor.execute, sql, params)

File "/srv/django-site/venv/lib/python3.4/site-packages/graphene_django/debug/sql/tracking.py" in _record
  103.             return method(sql, params)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/db/backends/utils.py" in execute
  80.             return super(CursorDebugWrapper, self).execute(sql, params)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/db/backends/utils.py" in execute
  65.                 return self.cursor.execute(sql, params)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/db/utils.py" in __exit__
  94.                 six.reraise(dj_exc_type, dj_exc_value, traceback)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/utils/six.py" in reraise
  685.             raise value.with_traceback(tb)

File "/srv/django-site/venv/lib/python3.4/site-packages/django/db/backends/utils.py" in execute
  65.                 return self.cursor.execute(sql, params)

Exception Type: ProgrammingError at /admin/fobi_contrib_plugins_form_handlers_db_store/savedformdataentry/
Exception Value: SELECT DISTINCT ON expressions must match initial ORDER BY expressions
LINE 1: SELECT DISTINCT ON ("db_store_savedformdataentry"."form_data...
                            ^
```

Just from the research I've done it seems like in Postgres DISTINCT requires the same argument used by ORDER BY if they are used together.

**Comment by barseghyanartur** [comment]
@jeremytiki:

Could you provide some more details? Which version of Postgres is it? I have just checked the living production site which uses fobi. Export of submitted form entries, as well as forms went well.

**Comment by jeremytiki** [comment]
@barseghyanartur My bad:

 PostgreSQL 9.2.23 on x86_64-redhat-linux-gnu, compiled by gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-16), 64-bit

**Comment by barseghyanartur** [comment]
I'm running it with no problems using the following stack:

- Debian
- PostgreSQL 9.4.18
- Python 3
- Django 1.11

However, I didn't test the export from admin. I'll check it soon.

In the meanwhile, you could use data export in the fobi's own management interface (since that one works).

**Comment by barseghyanartur** [comment]
Solved in version 0.13.3.
